lib/flat_search/olx_scraper.ex

```diff
@@ -1,28 +1,46 @@
-defmodule OlxScraper do
-  use Crawly.Spider
+defmodule FlatSearch.OlxScraper do
   alias FlatSearch.Flats
 
-  @impl Crawly.Spider
-  def base_url(), do: "https://www.olx.pl"
+  def run do
+    url = "https://www.olx.pl/nieruchomosci/mieszkania/wynajem/"
+    res = Crawly.fetch(url)
+    {:ok, document} = Floki.parse_document(res.body)
 
-  @impl Crawly.Spider
-  def init(), do: [start_urls: ["https://www.olx.pl/nieruchomosci/mieszkania/wynajem/"]]
+    # Run get_flat() for every page simultaneously
+    for n <- get_range_of_pages(document) do
+      Task.async(fn -> get_flats(url <> "?page=#{n}") end)
+    end
```

Comment:
`document |> get_range_of_pages() |> Enum.each(fn n -> Task.async(fn -> get_flats(url <> "?page=#{n}") end) end)`

Reply:
Personally, I find that less readable but I can change it

Reply:
Enum should be a little bit faster :P

Reply:
also you can change `url <> "?page=#{n}"` to `#{url}?page=#{n}` to shorten this